### ISSUE TITLE
Markdown line highlighting fails in v1.0.0-alpha.40

### ISSUE THREAD

**Issue by AngusFu** [vuejs/vuepress#1364]
- [x] I confirm that this is a issue rather than a question.

## Bug report

It seems that `v1.0.0-alpha.40` breaks markdown line highlighting feature.

**I believe this is caused by commit [f04adbf1](https://github.com/vuejs/vuepress/commit/f04adbf1eaef805736c1aa89a460b29cb59a2b17#diff-e8ddcc903cf9f678469a9dd38bf4b926)**, which changed `@vuepress/markdown-loader/index.js`. Since I replaced the content of `node_modules/@vuepress/markdown-loader/index.js` with code from previous version, everything turns normal.

#### Version

v1.0.0-alpha.40

#### Steps to reproduce

I created a simple repo: [vuepress-bug-report](https://github.com/AngusFu/vuepress-bug-report/tree/v1.0.0-alpha.40)

#### What is expected?

markdown line highlighting feature should works.

![image](https://user-images.githubusercontent.com/8180186/53478996-10632200-3ab3-11e9-843d-b266298589f4.png)

#### What is actually happening?

![image](https://user-images.githubusercontent.com/8180186/53478843-b82c2000-3ab2-11e9-8e78-3119dfebfaa9.png)

#### Other relevant information

None

**Comment by Shigma** [comment]
You are right. But I have no idea what caused the problem. 😭 Have you got any ideas?

**Comment by AngusFu** [comment]
@Shigma  The solution is simple. Either way:

- **remove** `md.parse = (src, env) => {`
- or, **recover** the LRU instance `cache` in `markdown-loader`

I have no idea why the whole compiling runs twice, with the requests being:

```
request:"${HOME}/vuepress-bug-report/node_modules/cache-loader/dist/cjs.js??ref--1-0!${HOME}/vuepress-bug-report/node_modules/vue-loader/lib/index.js??ref--1-1!${HOME}/vuepress-bug-report/node_modules/@vuepress/markdown-loader/index.js??ref--1-2!${HOME}/vuepress-bug-report/docs/README.md"
```

```
request:"${HOME}/vuepress-bug-report/node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/@vuepress/core/node_modules/.cache/vuepress","cacheIdentifier":"0954d92e-vue-loader-template"}!${HOME}/vuepress-bug-report/node_modules/vue-loader/lib/loaders/templateLoader.js??vue-loader-options!${HOME}/vuepress-bug-report/node_modules/cache-loader/dist/cjs.js??ref--1-0!${HOME}/vuepress-bug-report/node_modules/vue-loader/lib/index.js??ref--1-1!${HOME}/vuepress-bug-report/node_modules/@vuepress/markdown-loader/index.js??ref--1-2!${HOME}/vuepress-bug-report/docs/README.md?vue&type=template&id=c6f5370e&"
```

By rewriting `md.parse`, of course we can caching tokens, BUT `highlightLines.js` would  rewrite ([link](https://github.com/vuejs/vuepress/blob/5104a2854393d2a08ac3bbbdc90069bbef1af9fa/packages/%40vuepress/markdown/lib/highlightLines.js#L19)) `token.info` during the first compilation. Thus, when it comes to the second compilation process, it outputs the wrong result.

```
    const langName = rawInfo.replace(RE, '').trim()
    // ensure the next plugin get the correct lang.
    token.info = langName
```

**Comment by Shigma** [comment]
@AngusFu Thanks a lot!

To address this issue, I found a simple solution [here](https://github.com/vuejs/vuepress/pull/1369/files).

However, I believe **recover** the LRU instance `cache` is much safer for more potential problems. You are right. Accually, we have met such things [before](https://github.com/vuejs/vuepress/pull/1337/files#diff-4112e46619b1b18df2052a65cfab6677).

cc @ulivz

**Comment by meteorlxy** [comment]
And the markdown toc fails for the similar reason

**Comment by Shigma** [comment]
@meteorlxy Fixed at #1375
